opencl-aot/source/utils.cpp

```diff
@@ -220,7 +225,8 @@ getOpenCLPlatform(DeviceType Type) {
     auto Result =
         std::find(SupportedPlatformNames.begin(), SupportedPlatformNames.end(),
                   PlatformNameOnLoopIteration);
-    if (Result != SupportedPlatformNames.end()) {
+    if (Result != SupportedPlatformNames.end() &&                  // name match
+        !clFailed(std::get<2>(getOpenCLDevice(Platform, Type)))) { // type match
```

Comment:
Error code from ```getOpenCLDevice``` is ignored, it is better to return it as the last parameter of the tuple.

Reply:
`getOpenCLDevice(Platform, Type)` failure means that we cannot retrieve device ID of type `Type` for `Platform` on this iteration.

If this succeeds on a later iteration (platform of this type is found), we will return `CL_SUCCESS`.

If every iteration fails (platform of this type is never found), we will return `OPENCL_AOT_PLATFORM_NOT_FOUND`.

Reply:
My point is - we shouldn't return generic error code (```OPENCL_AOT_PLATFORM_NOT_FOUND```) in case if something failed on OpenCL side, we should report OpenCL error to the user, like 
```Failed to retrieve OpenCL device ID: -1 (CL_DEVICE_NOT_FOUND)```.